weld/vectorizer.rs

```diff
@@ -363,26 +436,22 @@ fn typed_expr(code: &str) -> TypedExpr {
 #[cfg(test)]
 pub fn has_vectorized_merge(expr: &TypedExpr) -> bool {
     let mut found = false;
-    expr.traverse(&mut |ref e| {
-        if let Merge { ref value, .. } = e.kind {
-            found |= value.ty.is_simd();
-        }
-    });
+    expr.traverse(&mut |ref e| if let Merge { ref value, .. } = e.kind {
+                           found |= value.ty.is_simd();
+                       });
     found
 }
 
 #[test]
 fn simple_merger() {
-    let mut e = typed_expr(
-        "|v:vec[i32]| result(for(v, merger[i32,+], |b,i,e| merge(b,e+1)))");
+    let mut e = typed_expr("|v:vec[i32]| result(for(v, merger[i32,+], |b,i,e| merge(b,e+1)))");
     vectorize(&mut e);
     assert!(has_vectorized_merge(&e));
 }
 
 #[test]
```

Comment:
Add a few unit tests for the dictmerger - I'd make the value being merged in something nontrivial in one of the tests (e.g., a binop or for loop) to make sure expressions aren't being copied